pyo3-ffi/src/cpython/abstract_.rs

```diff
@@ -42,14 +42,15 @@ extern "C" {
 }
 
 #[cfg(Py_3_8)]
-const PY_VECTORCALL_ARGUMENTS_OFFSET: Py_ssize_t =
-    1 << (8 * std::mem::size_of::<Py_ssize_t>() as Py_ssize_t - 1);
+const PY_VECTORCALL_ARGUMENTS_OFFSET: size_t =
+    1 << (8 * std::mem::size_of::<size_t>() as size_t - 1);
 
 #[cfg(Py_3_8)]
 #[inline(always)]
 pub unsafe fn PyVectorcall_NARGS(n: size_t) -> Py_ssize_t {
-    assert!(n <= (PY_SSIZE_T_MAX as size_t));
-    (n as Py_ssize_t) & !PY_VECTORCALL_ARGUMENTS_OFFSET
+    let real_n = n & !PY_VECTORCALL_ARGUMENTS_OFFSET;
+    assert!(real_n <= (PY_SSIZE_T_MAX as size_t));
+    real_n as Py_ssize_t
```

Comment:
I think nightly Clippy recently started to lint this code and recommending

```rust
n.try_into().unwrap()
```

instead.

Reply:
I don't see such a lint locally with nightly...

Reply:
The reference to lint was mainly to explain why I suggested the change now even though it is unrelated to the issue in question. I do think Clippy is right though and deferring to the standard library implementation is more readable and less error prone.

Reply:
Gotcha, will change it.